--- a/cpp/pointresponse/mwapoint.cc
+++ b/cpp/pointresponse/mwapoint.cc
@@ -102,15 +102,17 @@
   const telescope::MWA& mwatelescope =
       static_cast<const telescope::MWA&>(GetTelescope());
   casacore::MeasFrame frame(mwatelescope.GetArrayPosition(), time_epoch);
-  const casacore::Vector<double> itrf_coord(
-      {itrf_direction[0], itrf_direction[1], itrf_direction[2]});
-  const casacore::Quantum<casacore::Vector<double>> itrf(itrf_coord, "m");
-  const casacore::MDirection direction_itrf(itrf, casacore::MDirection::ITRF);
+
+  // itrf_direction is a unit direction cosine vector in ITRF, not meters.
+  const casacore::MDirection direction_itrf(
+      casacore::MVDirection(itrf_direction[0], itrf_direction[1],
+                            itrf_direction[2]),
+      casacore::MDirection::Ref(casacore::MDirection::ITRF, frame));
   casacore::MDirection::Convert measure_converter(
       casacore::MDirection::Ref(casacore::MDirection::ITRF, frame),
       casacore::MDirection::J2000);
   const casacore::Vector<double> j2000_dir =
-      measure_converter(itrf).getValue().getValue();
+      measure_converter(direction_itrf).getValue().getValue();
 
   ResponseDouble(beam_mode, result, j2000_dir[0], j2000_dir[1], freqs,
                  station_idx, 0);
